src/bluesearch/entrypoint/database/parse.py

```diff
@@ -0,0 +1,53 @@
+"""Parsing an article."""
+import argparse
+import json
+import pickle  # nosec
+
+from bluesearch.database.article import Article, CORD19ArticleParser
+
+
+def get_parser() -> argparse.ArgumentParser:
+    """Create a parser."""
+    parser = argparse.ArgumentParser(
+        description="Parse article.",
+    )
+    parser.add_argument(
+        "parser",
+        type=str,
+        help="""Parser class.""",
+    )
+    parser.add_argument(
+        "input_path",
+        type=str,
+        help="""Path to the file/directory to be parsed.""",
+    )
+    parser.add_argument(
+        "output_path",
+        type=str,
+        help="""Path where the parsed article is saved.""",
+    )
+    return parser
+
+
+def run(
+    *,
+    parser: str,
+    input_path: str,
+    output_path: str,
+) -> None:
+    """Parse an article.
+
+    Parameter description and potential defaults are documented inside of the
+    `get_parser` function.
+    """
+    if parser == "CORD19ArticleParser":
+        with open(input_path) as f:
+            parser_inst = CORD19ArticleParser(json.load(f))
+
+    else:
+        raise ValueError(f"Unsupported parser {parser}")
+
+    article = Article.parse(parser_inst)
+
+    with open(output_path, "wb") as f:  # type: ignore
+        pickle.dump(article, f)  # type: ignore
```

Comment:
Just curious, what's the problem with types here?

Reply:
```bash
src/bluesearch/entrypoint/database/parse.py:52: error: Incompatible types in assignment (expression has type "BinaryIO", variable has type "TextIO")  [assignment]
src/bluesearch/entrypoint/database/parse.py:53: error: Argument 2 to "dump" has incompatible type "TextIO"; expected "IO[bytes]"  [arg-type]
```

I did not really look into it too much:) If you know the recommended way around this feel free to share:)

Reply:
I think renaming `f` to a different name (that was not used in the same scope before) should fix this.

I guess there was another variable called `f` somewhere before this block of code that was of type `BinaryIO`. `mypy` does not allow re-assigning variables with a different type.

Reply:
You are right! I did not realize that the context manager actually lives on (is assigned to a variable) after being closed. Thank you! Fixed in 2a1801d6ba583ae15105b48e9e2b95dbd9c5bfd8